gcc/config/riscv/riscv.h

```diff
@@ -30,6 +30,9 @@ along with GCC; see the file COPYING3.  If not see
 /* Target CPU builtins.  */
 #define TARGET_CPU_CPP_BUILTINS() riscv_cpu_cpp_builtins (pfile)
 
+/* Zephyr toolchain wants strict aligned accesses by default */
+#define TARGET_DEFAULT_TARGET_FLAGS (TARGET_DEFAULT | MASK_STRICT_ALIGN)
```

Comment:
Any reason to not set `-mstrict-align` for RISC-V in the Zephyr build system?

That way, this problem can be also addressed for the users who are not using Zephyr SDK and we do not carry yet another Zephyr-specific patch.

Reply:
We need this when building *everything* -- including bits in the SDK (libc, libgcc and libstdc++). I think having the SDK set this internally will be the most reliable way to ensure that nothing gets compiled without this flag.

For other toolchains, I'm not sure how we'd ensure they had built everything correctly; the 'standard' risc-v ISA requires that the runtime environment support unaligned access, so they may fail when used to build Zephyr.

Adding this to the Zephyr flags would reduce the amount of code built incorrectly when not using the Zephyr SDK, so it's probably worth doing too.

Reply:
Fair enough.

Even if we end up adding the support for misaligned access exception/emulation, it would be better to not generate one in the first place for performance reasons.